lib/queues.js

```diff
@@ -260,13 +270,62 @@ class Queue {
   runOnce() {
     assert(!this._processing, 'Cannot call once while continuously processing jobs');
     const hasHandlers = (this._handlers.size > 0);
+
     if (hasHandlers) {
 
       debug('Waiting for jobs on queue %s', this.name);
-      return this._getClientPromise.then(client => this._reserveAndProcess(client));
+
+      if (process.env.NODE_ENV === 'test' && this._delayedJobs.size > 0) {
+
+        const jobIDs = getReadyJobIDs(this);
```

Comment:
We're already working with OOP, so maybe we can have "private" methods of the class (`_`-prefixed) instead of passing `this` as an argument?